### ISSUE TITLE
RS1034 is not reported if type is not exactly `SyntaxNode`.

### ISSUE THREAD

**Issue by carlreinke** [dotnet/roslyn-analyzers#7230]
### Analyzer

**Diagnostic ID**: [RS1034](https://github.com/dotnet/roslyn-analyzers/blob/v3.3.4/src/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.md#rs1034-prefer-iskind-for-checking-syntax-kinds): `Prefer 'IsKind' for checking syntax kinds`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.Analyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers/)

**Version**: 3.3.4

### Describe the bug

RS1034 is not reported if type is not exactly `SyntaxNode`.

### Steps To Reproduce

```C#
ExpressionSyntax node = null!;
bool noWarning = node.Kind() == SyntaxKind.NullLiteralExpression;
bool warning = ((SyntaxNode)node).Kind() == SyntaxKind.NullLiteralExpression;
```

### Expected behavior

RS1034 is reported on both instances of `Kind()` in example above.

### Actual behavior

RS1034 is only reported on the second instance of `Kind()` in example above.
